.github/workflows/ci.yml

```diff
@@ -343,7 +343,11 @@ jobs:
           # * Cache misses are quite costly:
           #   * 99% cache hits -> ~5 minutes to build
           #   * 20% cache hits -> ~15-20 minutes to build
-          CCACHE_MAXSIZE: 4G
+          # Cache restore on Macs is painfully slow and grinds to a complete
+          # halt somewhere around a cache size of 2GB. See
+          # https://github.com/actions/cache/issues/1115. Hopefully this is
+          # still large enough to be somewhat useful.
+          CCACHE_MAXSIZE: 1.5G
           CCACHE_COMPRESSLEVEL: 5
```

Comment:
I'm not optimistic about the cache being as useful as we'd like with the reduced size. Some cache is hopefully better than no cache though.

Reply:
Do you want to test this PR with cache writes enabled? The presubmit run with existing cache entries downloaded ~2795 MB: https://github.com/openxla/iree/actions/runs/4601599229/jobs/8129656448?pr=12905#step:7:247

Reply:
Yeah sure. I wasn't sure if it would actually work since it's coming from a fork, but we'll see

Reply:
So the good news is that fetching the cache only takes a minute. The bad news is that the build with this cache is not any faster than an uncached build. Ugh this is so annoying